### ISSUE TITLE
Freeze when trying to compile gallery example with skia and wayland

### ISSUE THREAD

**Issue by qarmin** [slint-ui/slint#3659]
Slint 1.3.0 - 400404623d24df1520f9f836e845d3e0a2fb9f85

When trying to compile app with skia and wayland on armv7, then I got freeze on compilation stage
```
   Compiling clru v0.6.1
   Compiling scoped-tls-hkt v0.1.4
   Compiling pin-weak v1.1.0
   Compiling smithay-clipboard v0.6.6
   Compiling i-slint-backend-winit v1.3.0 (/project/internal/backends/winit)
   Compiling glow v0.12.3
   Compiling copypasta v0.8.2
   Compiling i-slint-backend-selector v1.3.0 (/project/internal/backends/selector)
   Compiling i-slint-core v1.3.0 (/project/internal/core)
   Compiling toml_edit v0.20.2
   Compiling send_wrapper v0.6.0
   Compiling slint-build v1.3.0 (/project/api/rs/build)
   Compiling gallery v1.3.0 (/project/examples/gallery)
   Compiling slint-macros v1.3.0 (/project/api/rs/macros)
    Building [=======================> ] 446/454: skia-bindings(build) 
```
I waited 15 minutes and nothing changed, cpu usage is almost 0%

Steps to reproduce
```
cd slint/examples/gallery
nano Cargo.toml

# Replace slint line into 
# slint = { path = "../../api/rs/slint", default-features=false, features=["gettext", "backend-winit-wayland", "renderer-skia"] }

cross build --target armv7-unknown-linux-gnueabihf --release
````
This change:
```
slint = { path = "../../api/rs/slint", default-features=false, features=["gettext", "backend-winit-wayland", "renderer-software", "compat-1-2"] }
```
~compiles fine I got error(probably my fault)~ - yes this was my error, opening app via `WAYLAND_DISPLAY=wayland-0 ./gallery` fixed problem
```
thread 'main' panicked at /cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.28.7/src/platform_impl/linux/mod.rs:757:9:
Failed to initialize any backend! Wayland status: NoCompositorListening X11 status: "backend disabled"
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Aborted
```

Default build compiles fine, but I cannot open due error:
```
thread 'main' panicked at /cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.28.7/src/platform_impl/linux/mod.rs:757:9:
Failed to initialize any backend! Wayland status: NoCompositorListening X11 status: LibraryOpenError(OpenError { kind: Library, detail: "opening library failed (libX11.so.6: cannot open shared object file: No such file or directory); opening library failed (libX11.so: cannot open shared object file: No such file or directory)" })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by tronical** [comment]
Thanks for your report! This is rather odd. When the build hangs, could you enter into the container with `docker exec -it $container bash` and see with `ps -ef` what the processes are that are sitting around?

**Comment by tronical** [comment]
One thing that's worth nothing is that this kind of build will download the skia sources from the network (as the pre-built skia binaries don't include armv7). That might be the cause for what seems like nothing is happening: The download through the network might be taking time.

**Comment by qarmin** [comment]
Looks like you are right, network usage is 1MB/s most of the time 
```
UID          PID    PPID  C STIME TTY          TIME CMD
1000           1       0  0 08:33 pts/0    00:00:00 sh -c PATH=$PATH:/rust/bin cargo build --target armv7-unknown-linux-gnueabihf --release
1000           7       1  1 08:33 pts/0    00:00:01 cargo build --target armv7-unknown-linux-gnueabihf --release
1000        2358       0  0 08:33 pts/1    00:00:00 /bin/bash
1000        4219       7  0 08:34 pts/0    00:00:00 /target/release/build/skia-bindings-f774deedde436b79/build-script-build
1000        4275    4219  0 08:34 pts/0    00:00:00 python3 skia/tools/git-sync-deps
1000        4290    4275  0 08:34 pts/0    00:00:00 git clone --quiet --no-checkout https://chromium.googlesource.com/chromium/deps/icu.git skia/third_party/externals/icu
1000        4306    4290  1 08:34 pts/0    00:00:01 /usr/lib/git-core/git-remote-https origin https://chromium.googlesource.com/chromium/deps/icu.git
1000        4344    4306  0 08:34 pts/0    00:00:00 /usr/lib/git-core/git fetch-pack --stateless-rpc --stdin --lock-pack --thin --check-self-contained-and-connected --cloning --no-progress https:
1000        4352    4344  3 08:34 pts/0    00:00:05 /usr/lib/git-core/git index-pack --stdin --fix-thin --keep=fetch-pack 4344 on 2e89ef3d13dd --check-self-contained-and-connected --pack_header=2
1000        4497    2358  0 08:36 pts/1    00:00:00 ps -ef
```

**Comment by tronical** [comment]
I'm closing this as out of scope for now - not sure there's much that we can do on the Slint side here. It builds for me and network connectivity is indeed required (like for most builds with rust when crates are downloaded).
